Tests/src/PublicAPITests/MaterialTests.cpp

```diff
@@ -0,0 +1,315 @@
+#include "CSP/Systems/Assets/AssetSystem.h"
+#include "CSP/Systems/SystemsManager.h"
+#include "SpaceSystemTestHelpers.h"
+#include "TestHelpers.h"
+#include "UserSystemTestHelpers.h"
+
+#include "gtest/gtest.h"
+
+using namespace csp::systems;
+
+namespace
+{
+
+bool RequestPredicate(const csp::systems::ResultBase& Result)
+{
+	return Result.GetResultCode() != csp::systems::EResultCode::InProgress;
+}
+
+} // namespace
+
+void CreateMaterial(AssetSystem* AssetSystem, const csp::common::String& Name, const csp::common::String& SpaceId, GLTFMaterial& OutMaterial)
+{
+	auto [Result] = AWAIT_PRE(AssetSystem, CreateMaterial, RequestPredicate, Name, SpaceId);
+
+	EXPECT_EQ(Result.GetResultCode(), csp::systems::EResultCode::Success);
+
+	const GLTFMaterial& Material = Result.GetGLTFMaterial();
+
+	EXPECT_EQ(Material.GetName(), Name);
+
+	OutMaterial = Result.GetGLTFMaterial();
+}
+
+void UpdateMaterial(AssetSystem* AssetSystem, const GLTFMaterial& Material)
+{
+	auto [Result] = AWAIT_PRE(AssetSystem, UpdateMaterial, RequestPredicate, Material);
+
+	EXPECT_EQ(Result.GetResultCode(), csp::systems::EResultCode::Success);
+}
+
+void DeleteMaterial(AssetSystem* AssetSystem, const csp::common::String& AssetCollectionId, const csp::common::String& AssetId)
+{
+	auto [Result] = AWAIT_PRE(AssetSystem, DeleteMaterial, RequestPredicate, AssetCollectionId, AssetId);
+
+	EXPECT_EQ(Result.GetResultCode(), csp::systems::EResultCode::Success);
+}
+
+void GetMaterials(AssetSystem* AssetSystem, const csp::common::String& SpaceId, csp::common::Array<GLTFMaterial>& OutMaterials)
+{
+	auto [Result] = AWAIT_PRE(AssetSystem, GetMaterials, RequestPredicate, SpaceId);
+
+	EXPECT_EQ(Result.GetResultCode(), csp::systems::EResultCode::Success);
+
+	OutMaterials = Result.GetGLTFMaterials();
+}
+
+void GetMaterial(AssetSystem* AssetSystem,
+				 const csp::common::String& AssetCollectionId,
+				 const csp::common::String& AssetId,
+				 GLTFMaterial& OutMaterial)
+{
+	auto [Result] = AWAIT_PRE(AssetSystem, GetMaterial, RequestPredicate, AssetCollectionId, AssetId);
+
+	EXPECT_EQ(Result.GetResultCode(), csp::systems::EResultCode::Success);
+
+	OutMaterial = Result.GetGLTFMaterial();
+}
+
+#if RUN_ALL_UNIT_TESTS || RUN_MATERIAL_TESTS || RUN_MATERIAL_CREATEMATERIAL_TEST
+
+CSP_PUBLIC_TEST(CSPEngine, MaterialTests, CreateMaterialTest)
+{
+	SetRandSeed();
+
+	auto& SystemsManager = ::SystemsManager::Get();
+	auto* UserSystem	 = SystemsManager.GetUserSystem();
+	auto* SpaceSystem	 = SystemsManager.GetSpaceSystem();
+	auto* AssetSystem	 = SystemsManager.GetAssetSystem();
+
+	// Log in
+	csp::common::String UserId;
+	LogIn(UserSystem, UserId);
+
+	// Create space to associate a material with
+	::Space Space;
+	CreateDefaultTestSpace(SpaceSystem, Space);
+
+	GLTFMaterial CreatedMaterial;
+	CreateMaterial(AssetSystem, "TestMaterial", Space.Id, CreatedMaterial);
+
+	// Cleanup
+	DeleteMaterial(AssetSystem, CreatedMaterial.GetAssetCollectionId(), CreatedMaterial.GetAssetId());
```

Comment:
[Consider] I would have expected a test here, to ensure the material was actually created (and deleted)

Reply:
I believe this is done is the GetMaterial and the new DeleteMaterial tests

Reply:
Ah of course, naming heh.